engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/PreviewWorldScreen.java

```diff
@@ -87,22 +90,39 @@
 
     private ModuleEnvironment environment;
 
+    private final Texture texture;
+
+    private boolean triggerUpdate;
+
+    public PreviewWorldScreen() {
+        int imgWidth = 384;
+        int imgHeight = 384;
+        ByteBuffer buffer = ByteBuffer.allocateDirect(imgWidth * imgHeight * Integer.BYTES);
+        ByteBuffer[] data = new ByteBuffer[]{buffer};
+        AssetUri uri = new AssetUri(AssetType.TEXTURE, "engine:terrainPreview");
+        TextureData texData = new TextureData(imgWidth, imgHeight, data, Texture.WrapMode.CLAMP, Texture.FilterMode.LINEAR);
+        texture = Assets.generateAsset(uri, texData, Texture.class);
+    }
+
     @Override
     public void onOpened() {
         super.onOpened();
 
-        CoreRegistry.put(WorldGeneratorPluginLibrary.class, new TempWorldGeneratorPluginLibrary());
         SimpleUri worldGenUri = config.getWorldGeneration().getDefaultGenerator();
+        Name moduleName = worldGenUri.getModuleName();
 
         try {
             DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
-            Name moduleName = worldGenUri.getModuleName();
             ResolutionResult result = resolver.resolve(moduleName);
             if (result.isSuccess()) {
                 environment = moduleManager.loadEnvironment(result.getModules(), false);
+                CoreRegistry.put(WorldGeneratorPluginLibrary.class, new TempWorldGeneratorPluginLibrary(environment));
+                CoreRegistry.get(AssetManager.class).setEnvironment(environment);
```

Comment:
I am currently trying to remove CoreRegistry usage. The AssetManager can also be obtained via depenendcy injection. e.g. by adding a private AssetManager field with an @In annotation.

The registration can be done after #1721 is merged by obtaining a Context instance via dependency injection. It offers a put method like CoreRegistry. Although I find a bit strange of a location for registering objects.